data/txHandler_test.go

```diff
@@ -1407,11 +1545,22 @@ func runHandlerBenchmarkWithBacklog(maxGroupSize, tps int, invalidRate float32,
 		}()
 	}
 
-	// Prepare the transactions
-	signedTransactionGroups, badTxnGroups := makeSignedTxnGroups(b.N, numUsers, maxGroupSize, invalidRate, addresses, secrets)
+	// Prepare 1000 transactions
+	genTCount := 1000
+	if b.N < genTCount {
+		genTCount = b.N
+	}
+	var signedTransactionGroups [][]transactions.SignedTxn
+	var badTxnGroups map[uint64]interface{}
+	if msigSize == 1 {
+		signedTransactionGroups, badTxnGroups = makeSignedTxnGroups(genTCount, numUsers, maxGroupSize, invalidRate, addresses, secrets)
+	} else {
+		signedTransactionGroups, badTxnGroups, err = makeMsigSignedTxnGroups(genTCount, numUsers, msigSize, maxGroupSize, invalidRate, addresses, secrets)
+		require.NoError(b, err)
+	}
```

Comment:
let's make a better abstraction here - an interface to produce transactions as needed. I'll submit a PR against your with lsigs added.

Reply:
that will be great! thanks.